File: android/AlpineApp/app/src/main/java/com/alpine/app/data/transport/QueryTransport.kt
package com.alpine.app.data.transport

import com.alpine.app.data.model.QueryRequest
import com.alpine.app.data.model.QueryResponse
import com.alpine.app.data.model.QueryResultsResponse
import com.alpine.app.data.model.QueryStatusResponse

interface QueryTransport {
    suspend fun startQuery(request: QueryRequest): Result<QueryResponse>
    suspend fun getQueryStatus(queryId: Long): Result<QueryStatusResponse>
    suspend fun getQueryResults(queryId: Long): Result<QueryResultsResponse>
    suspend fun cancelQuery(queryId: Long): Result<Unit>
    fun shutdown()
}

File: ios/AlpineApp/AlpineApp/Transport/QueryTransport.swift
import Foundation

protocol QueryTransport: Sendable {
    func startQuery(_ request: QueryRequest) async throws -> QueryResponse
    func getQueryStatus(_ queryId: Int64) async throws -> QueryStatusResponse
    func getQueryResults(_ queryId: Int64) async throws -> QueryResultsResponse
    func cancelQuery(_ queryId: Int64) async throws
    func shutdown()
}
